src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,28 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception e) {
+            transaction.rollback();
+            throw new DataProcessingException("Can't add movie " + movie, e);
```

Comment:
The exception message should be more specific. Include the movie's title or ID to make it easier to identify the exact movie that failed to be added.